### PR TITLE
Port a few modules up to 1.21

### PR DESCRIPTION
**Pull request by OroArmor**
Also makes the DISABLE files per module, making it less of a diff to enable them.

### CODE REVIEW

**Comment by eristhea** [comment]
the deletions?? you're cooking??

**Comment by OroArmor** [comment]
> the deletions?? you're cooking??

Nope! Just using Mixin Extras

File: library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/MutablePack.java [hunk 1]
```diff
@@ -293,6 +294,14 @@ default void putImage(Identifier id, Supplier<NativeImage> imageSupplier) {
 		}));
 	}
 
+	/**
+	 * Adds a pack overlay to the mutable pack.
+	 *
+	 * @param overlay the overlay name
+	 * @param pack    the pack
+	 */
+	void putOverlay(@NotNull String overlay, ResourcePack pack);
```
Comment (OroArmor) [inline library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/MutablePack.java:303]: @ix0rai @TheGlitch76 Im not too sure about this. Should it be a `ResourcePack`, where any pack can be part of the mutable pack, or does `Consumer<ResourcePack>` make more sense where the `MutablePack` implementation creates the pack in the way that it wants? Both options have their advantages.

**Comment by eristhea** [comment]
@OroArmor can you compile a list of modules that you've ported in this PR?

**Comment by OroArmor** [comment]
The list is:
- Everything

Ported already (thanks glitch):
- base
- crash

**Comment by eristhea** [comment]
oh perf ok

File: library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/MutablePack.java (hunk 1 above)
Comment (TheGlitch76) [reply library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/api/MutablePack.java:303]: I have zero expertise in this area, it's really up to you...

**Review by TheGlitch76 — APPROVED** [review]

**Pull request merged**